### PR TITLE
ISIC2018_segmentation-s4824208

### PR DESCRIPTION
**Pull request by Snorrebs**
Name: Snorre Sveen
SID:s 4824208
Taks1: Segmentation of the ISIC2018 with UNet

This pull request contains the implementation of a UNet model designed to preform segmentation on the ISIC2018 dataset.
For more information about the model and the results, see the README file.

README.md--Report for the task completed
Report_img--Contains images for the README report.
train.py-- Used to train and test the UNET model
predict.py-- Prints plots of preformance on testingset + prediction images compeard with thruths
dataset.py-- Loads the dataset and applies image transforms
utils.py-- Contains dice loss function
models.py--  Defines the model arcitchture for the UNet model

To train the model:
1.) Go to dataset.py
     Add path to training set in line 84
     Add path to truths for training set in line 87
     Add path to test set in line 90 and test set truths 91
2.) Run train.py
3.) The model will be saved in same folder as train.py with the name "UNet_model.pt"
4.) Now its possible to run predict.py to get a plot of the scores on the test set and some example images.

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Well organized files.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.79, but no evidence to support this score. Tending to train the model for only 2 epochs. No predicted Mask output. 
- Correctly using PyTorch to construct the Improved U-Net models and functions.
- Sufficient comment in the code, but need some references.
- Easy difficulty problem, no higher difficulty marks.
- The format of the readme.md is not reasonable.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe OK, Markdown no used well -1, missing usage -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
